Add the new git-clone-oci-ta Task

lcarva opened the pull request:
This pull request introduces a new Tekton Task, `git-clone-oci-ta`. This is a variation of the `0.1/git-clone` Task. The main difference is that instead of cloning to a workspace, this Task stores the cloned repository as a Trusted Artifact in an OCI registry.

The Task is not currently used in any of the default Pipelines. This will happen later on.

I tested this Task on a kind cluster like this:

```
🐚 oc -n default apply -f task/git-clone-oci-ta/0.1/git-clone-oci-ta.yaml 
task.tekton.dev/git-clone-oci-ta configured

🐚 tkn -n default task start git-clone-oci-ta --showlog --param url=https://github.com/lcarva/festoji.git --param revision=master --param ociStorage=quay.io/lucarval/ec-550:latest --use-param-defaults --skip-optional-workspace

TaskRun started: git-clone-oci-ta-run-lp7dx
Waiting for logs to be available...
[clone] {"level":"info","ts":1715375774.0440822,"caller":"git/git.go:178","msg":"Successfully cloned https://github.com/lcarva/festoji.git @ 848edc452ccbc6d42ec56c2807eef2f49e754c5e (grafted, HEAD, origin/master) in path /var/source"}
[clone] {"level":"info","ts":1715375774.0673118,"caller":"git/git.go:217","msg":"Successfully initialized and updated submodules in path /var/source"}

[symlink-check] Running symlink check

[create-trusted-artifact] gzip: warning: GZIP environment variable is deprecated; use an alias or script
[create-trusted-artifact] Prepared artifact from /var/source (sha256:36282ab2ac313d04362043bc12e8f0f4872ea88425ce696fbb0a45819f2e52d7)
[create-trusted-artifact] Using token for quay.io
[create-trusted-artifact] Uploading 36282ab2ac31 sourceArtifact
[create-trusted-artifact] Uploaded  36282ab2ac31 sourceArtifact
[create-trusted-artifact] Pushed [registry] quay.io/lucarval/ec-550:latest
[create-trusted-artifact] Digest: sha256:fd907433bcd9dd790d792df40c104fb1284357aacc1736edfe23ce030afb086d
[create-trusted-artifact] Artifacts created
[create-trusted-artifact] 
```

Resolves [EC-550](https://issues.redhat.com/browse/EC-550)

lcarva commented:
Tip for reviewers: the first commit in this PR is a plain copy of the `0.1/git-clone` Task. This should make it easier to identify the actual changes in the new Task.

lcarva commented on task/git-clone-oci-ta/0.1/git-clone-oci-ta.yaml:
```diff
@@ -0,0 +1,291 @@
+      fi
+
+  - name: create-trusted-artifact
+    # TODO: Is there a released version of this?
```
Unclear if this image is available anywhere else ATM.

zregvart replied:
It used to be available [here](https://quay.io/repository/redhat-appstudio/build-trusted-artifacts?tab=tags&tag=latest), see also https://gitlab.cee.redhat.com/releng/konflux-release-data/-/merge_requests/213

lcarva replied:
Release was fixed. Now using `quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:4e39fb97f4444c2946944482df47b39c5bbc195c54c6560b0647635f553ab23d`

lcarva commented on task/git-clone-oci-ta/OWNERS:
```diff
@@ -0,0 +1 @@
+Stonesoup Build Team
```
Not sure if this is correct.

simonbaird replied:
Out of scope for this PR I'd say.

simonbaird commented on task/git-clone-oci-ta/0.1/README.md:
```diff
@@ -11,27 +11,28 @@ The git-clone Task will clone a repo from the provided url into the output Works
-|gitInitImage|Deprecated. Has no effect. Will be removed in the future.|""|false|
-|userHome|Absolute path to the user's home directory. Set this explicitly if you are running the image as a non-root user or have overridden the gitInitImage param with an image containing custom user configuration. |/tekton/home|false|
-|enableSymlinkCheck|Check symlinks in the repo. If they're pointing outside of the repo, the build will fail.|true|false|
+|verbose|Log the commands that are executed during `git-clone`'s operation.|false|false|
```
Intentionally changing default here I assume?

lcarva replied:
I re-generated this file with the `./hack/generate-readme.sh` script. It turns out that the README.md of the original git-clone task is out of date. The implementation itself has the default value of `false`.

lcarva commented on task/git-clone-oci-ta/0.1/git-clone-oci-ta.yaml:
```diff
@@ -151,6 +146,8 @@ spec:
         mountPath: /mnt/trusted-ca
         readOnly: true
+      - name: source
+        mountPath: /var/source
```
That won't work because `CHECKOUT_DIR` is not a parameter. It's an environment variable. I don't see a way of using that data [here](https://tekton.dev/docs/pipelines/variables/). We could make CHECKOUT_DIR a parameter and then simply use that in both places. But turning this into a parameter, makes it user visible which is not really the intention. Let's leave it as is.

chmeliik commented on task/git-clone-oci-ta/0.1/README.md:
```diff
@@ -11,27 +11,28 @@ The git-clone Task will clone a repo from the provided url into the output Works
+|caTrustConfigMapName|The name of the ConfigMap to read CA bundle data from.|trusted-ca|false|
+|caTrustConfigMapKey|The name of the key in the ConfigMap that contains the CA bundle data.|ca-bundle.crt|false|
+|ociStorage|The OCI repository where the clone repository will be stored.||true|
+|imageExpiresAfter|Expiration date for the artifacts created in the OCI repository.|""|false|
```
Nitpick: consider changing to something like `ociArtifactExpiresAfter` to tie this to the `ociStorage` param. `IMAGE_EXPIRES_AFTER` is a buildah param for the "main" image, so this could get confusing

lcarva replied:
Good idea! Done.

chmeliik reviewed (approved):
LGTM with one nitpick

lcarva commented:
/retest

lcarva commented:
/retest

Seems like an unrelated issue:
```
  PipelineRun devfile-sample-python-basic-pons-c257z reason: Running
  PipelineRun devfile-sample-python-basic-pons-c257z reason: Running
  attempt 3/3: PipelineRun "devfile-sample-python-basic-pons-c257z" failed: context deadline exceeded
  [FAILED] in [It] - /github.com/redhat-appstudio/e2e-tests/tests/build/build_templates.go:200 @ 05/14/24 16:24:49.907
  << Timeline

  [FAILED] Expected success, but got an error:
      <context.deadlineExceededError>: 
      context deadline exceeded
      {}
  In [It] at: /github.com/redhat-appstudio/e2e-tests/tests/build/build_templates.go:200 @ 05/14/24 16:24:49.907
```

lcarva commented:
/retest

lcarva commented:
Side note: these changes should have no impact on the existing tasks and pipelines. e2e-tests errors are likely unrelated.

lcarva commented:
Weird e2e failure...

```
• [FAILED] [1886.237 seconds]
[build-service-suite Build templates E2E test] HACBS pipelines [It] should eventually finish successfully for component with Git source URL https://github.com/redhat-appstudio-qe/devfile-sample-python-basic [build, build-templates, HACBS, pipeline, build-templates-e2e]
/github.com/redhat-appstudio/e2e-tests/tests/build/build_templates.go:196

  Timeline >>
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-hkfn5 reason: Failed
  attempt 1/3: PipelineRun "devfile-sample-python-basic-fgrv-hkfn5" failed: 
  init container prepare: 
  2024/05/14 17:51:05 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 17:51:09 Decoded script /tekton/scripts/script-0-z7gz9
  2024/05/14 17:51:09 Decoded script /tekton/scripts/script-1-dxl6h

  container step-clone: 
  {"level":"info","ts":1715709115.2848508,"caller":"git/git.go:178","msg":"Successfully cloned https://github.com/redhat-appstudio-qe/devfile-sample-python-basic @ 47fc22092005aabebce233a9b6eab994a8152bbd (grafted, HEAD) in path /workspace/output/source"}
  {"level":"info","ts":1715709115.3530369,"caller":"git/git.go:217","msg":"Successfully initialized and updated submodules in path /workspace/output/source"}

  container step-symlink-check: 
  Running symlink check

  init container prepare: 
  2024/05/14 17:48:45 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 17:48:46 Decoded script /tekton/scripts/script-0-q4rwm

  container step-init: 
  Build Initialize: quay.io/redhat-user-workloads/build-templates-e2e/test-app-ref-09d6e65894f8cd1a13ba4019d5c0342e19ec773d/devfile-sample-python-basic-fgrv:build-d79c9-1715708828

  Determine if Image Already Exists

  init container prepare: 
  2024/05/14 17:54:46 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 17:54:47 Decoded script /tekton/scripts/script-0-qddz9

  init container working-dir-initializer: 

  container step-sast-snyk-check: 
  {"result":"SKIPPED","timestamp":"1715709292","note":"Task sast-snyk-check skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key snyk_token containing the Snyk token by following the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)","namespace":"default","successes":0,"failures":0,"warnings":0}

  init container prepare: 
  2024/05/14 17:59:58 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 17:59:59 Decoded script /tekton/scripts/script-0-2zxtp

  container step-appstudio-summary: 

  Build Summary:

  Build repository: https://github.com/redhat-appstudio-qe/devfile-sample-python-basic?rev=47fc22092005aabebce233a9b6eab994a8152bbd

  End Summary
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: ResolvingPipelineRef
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: ResolvingPipelineRef
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-7nqnn reason: Failed
  attempt 2/3: PipelineRun "devfile-sample-python-basic-fgrv-7nqnn" failed: 
  init container prepare: 
  2024/05/14 18:09:11 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:09:14 Decoded script /tekton/scripts/script-0-8gqrf
  2024/05/14 18:09:14 Decoded script /tekton/scripts/script-1-vz8b8

  container step-clone: 
  {"level":"info","ts":1715710276.7260363,"caller":"git/git.go:178","msg":"Successfully cloned https://github.com/redhat-appstudio-qe/devfile-sample-python-basic @ 47fc22092005aabebce233a9b6eab994a8152bbd (grafted, HEAD) in path /workspace/output/source"}
  {"level":"info","ts":1715710276.8273754,"caller":"git/git.go:217","msg":"Successfully initialized and updated submodules in path /workspace/output/source"}

  container step-symlink-check: 
  Running symlink check

  init container prepare: 
  2024/05/14 18:04:43 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:05:05 Decoded script /tekton/scripts/script-0-fpmqb

  container step-init: 
  Build Initialize: quay.io/redhat-user-workloads/build-templates-e2e/test-app-ref-09d6e65894f8cd1a13ba4019d5c0342e19ec773d/devfile-sample-python-basic-fgrv:build-f4352-1715709757

  Determine if Image Already Exists

  init container prepare: 
  2024/05/14 18:12:08 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:12:09 Decoded script /tekton/scripts/script-0-f5lpf

  init container working-dir-initializer: 

  container step-sast-snyk-check: 
  {"result":"SKIPPED","timestamp":"1715710333","note":"Task sast-snyk-check skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key snyk_token containing the Snyk token by following the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)","namespace":"default","successes":0,"failures":0,"warnings":0}

  init container prepare: 
  2024/05/14 18:13:15 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:13:16 Decoded script /tekton/scripts/script-0-sk25j

  container step-appstudio-summary: 

  Build Summary:

  Build repository: https://github.com/redhat-appstudio-qe/devfile-sample-python-basic?rev=47fc22092005aabebce233a9b6eab994a8152bbd

  End Summary
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: 
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: ResolvingPipelineRef
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Running
  PipelineRun devfile-sample-python-basic-fgrv-x6q8m reason: Failed
  attempt 3/3: PipelineRun "devfile-sample-python-basic-fgrv-x6q8m" failed: 
  init container prepare: 
  2024/05/14 18:16:18 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:16:19 Decoded script /tekton/scripts/script-0-tdqkl
  2024/05/14 18:16:19 Decoded script /tekton/scripts/script-1-fjq6l

  container step-clone: 
  {"level":"info","ts":1715710595.317364,"caller":"git/git.go:178","msg":"Successfully cloned https://github.com/redhat-appstudio-qe/devfile-sample-python-basic @ 47fc22092005aabebce233a9b6eab994a8152bbd (grafted, HEAD) in path /workspace/output/source"}
  {"level":"info","ts":1715710595.354573,"caller":"git/git.go:217","msg":"Successfully initialized and updated submodules in path /workspace/output/source"}

  container step-symlink-check: 
  Running symlink check

  init container prepare: 
  2024/05/14 18:15:28 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:15:30 Decoded script /tekton/scripts/script-0-z4kx8

  container step-init: 
  Build Initialize: quay.io/redhat-user-workloads/build-templates-e2e/test-app-ref-09d6e65894f8cd1a13ba4019d5c0342e19ec773d/devfile-sample-python-basic-fgrv:build-4fcbf-1715710481

  Determine if Image Already Exists

  init container prepare: 
  2024/05/14 18:17:31 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:17:32 Decoded script /tekton/scripts/script-0-bb54t

  init container working-dir-initializer: 

  container step-sast-snyk-check: 
  {"result":"SKIPPED","timestamp":"1715710665","note":"Task sast-snyk-check skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key snyk_token containing the Snyk token by following the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)","namespace":"default","successes":0,"failures":0,"warnings":0}

  init container prepare: 
  2024/05/14 18:18:19 Entrypoint initialization

  init container place-scripts: 
  2024/05/14 18:18:20 Decoded script /tekton/scripts/script-0-f2m9s

  container step-appstudio-summary: 

  Build Summary:

  Build repository: https://github.com/redhat-appstudio-qe/devfile-sample-python-basic?rev=47fc22092005aabebce233a9b6eab994a8152bbd

  End Summary
  [FAILED] in [It] - /github.com/redhat-appstudio/e2e-tests/tests/build/build_templates.go:200 @ 05/14/24 18:19:17.576
  << Timeline

  [FAILED] Expected success, but got an error:
      <*errors.errorString | 0xc0009506c0>: 
      
      init container prepare: 
      2024/05/14 18:16:18 Entrypoint initialization
      
      init container place-scripts: 
      2024/05/14 18:16:19 Decoded script /tekton/scripts/script-0-tdqkl
      2024/05/14 18:16:19 Decoded script /tekton/scripts/script-1-fjq6l
      
      container step-clone: 
      {"level":"info","ts":1715710595.317364,"caller":"git/git.go:178","msg":"Successfully cloned https://github.com/redhat-appstudio-qe/devfile-sample-python-basic @ 47fc22092005aabebce233a9b6eab994a8152bbd (grafted, HEAD) in path /workspace/output/source"}
      {"level":"info","ts":1715710595.354573,"caller":"git/git.go:217","msg":"Successfully initialized and updated submodules in path /workspace/output/source"}
      
      container step-symlink-check: 
      Running symlink check
      
      init container prepare: 
      2024/05/14 18:15:28 Entrypoint initialization
      
      init container place-scripts: 
      2024/05/14 18:15:30 Decoded script /tekton/scripts/script-0-z4kx8
      
      container step-init: 
      Build Initialize: quay.io/redhat-user-workloads/build-templates-e2e/test-app-ref-09d6e65894f8cd1a13ba4019d5c0342e19ec773d/devfile-sample-python-basic-fgrv:build-4fcbf-1715710481
      
      Determine if Image Already Exists
      
      init container prepare: 
      2024/05/14 18:17:31 Entrypoint initialization
      
      init container place-scripts: 
      2024/05/14 18:17:32 Decoded script /tekton/scripts/script-0-bb54t
      
      init container working-dir-initializer: 
      
      container step-sast-snyk-check: 
      {"result":"SKIPPED","timestamp":"1715710665","note":"Task sast-snyk-check skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key snyk_token containing the Snyk token by following the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)","namespace":"default","successes":0,"failures":0,"warnings":0}
      
      init container prepare: 
      2024/05/14 18:18:19 Entrypoint initialization
      
      init container place-scripts: 
      2024/05/14 18:18:20 Decoded script /tekton/scripts/script-0-f2m9s
      
      container step-appstudio-summary: 
      
      Build Summary:
      
      Build repository: https://github.com/redhat-appstudio-qe/devfile-sample-python-basic?rev=47fc22092005aabebce233a9b6eab994a8152bbd
      
      End Summary
      
      {
          s: "\ninit container prepare: \n2024/05/14 18:16:18 Entrypoint initialization\n\ninit container place-scripts: \n2024/05/14 18:16:19 Decoded script /tekton/scripts/script-0-tdqkl\n2024/05/14 18:16:19 Decoded script /tekton/scripts/script-1-fjq6l\n\ncontainer step-clone: \n{\"level\":\"info\",\"ts\":1715710595.317364,\"caller\":\"git/git.go:178\",\"msg\":\"Successfully cloned https://github.com/redhat-appstudio-qe/devfile-sample-python-basic @ 47fc22092005aabebce233a9b6eab994a8152bbd (grafted, HEAD) in path /workspace/output/source\"}\n{\"level\":\"info\",\"ts\":1715710595.354573,\"caller\":\"git/git.go:217\",\"msg\":\"Successfully initialized and updated submodules in path /workspace/output/source\"}\n\ncontainer step-symlink-check: \nRunning symlink check\n\ninit container prepare: \n2024/05/14 18:15:28 Entrypoint initialization\n\ninit container place-scripts: \n2024/05/14 18:15:30 Decoded script /tekton/scripts/script-0-z4kx8\n\ncontainer step-init: \nBuild Initialize: quay.io/redhat-user-workloads/build-templates-e2e/test-app-ref-09d6e65894f8cd1a13ba4019d5c0342e19ec773d/devfile-sample-python-basic-fgrv:build-4fcbf-1715710481\n\nDetermine if Image Already Exists\n\ninit container prepare: \n2024/05/14 18:17:31 Entrypoint initialization\n\ninit container place-scripts: \n2024/05/14 18:17:32 Decoded script /tekton/scripts/script-0-bb54t\n\ninit container working-dir-initializer: \n\ncontainer step-sast-snyk-check: \n{\"result\":\"SKIPPED\",\"timestamp\":\"1715710665\",\"note\":\"Task sast-snyk-check skipped: If you wish to use the Snyk code SAST task, please create a secret name snyk-secret with the key snyk_token containing the Snyk token by following the steps given [here](https://redhat-appstudio.github.io/docs.appstudio.io/Documentation/main/how-to-guides/testing_applications/enable_snyk_check_for_a_product/)\",\"namespace\":\"default\",\"successes\":0,\"failures\":0,\"warnings\":0}\n\ninit container prepare: \n2024/05/14 18:18:19 Entrypoint initialization\n\ninit container place-scripts: \n2024/05/14 18:18:20 Decoded script /tekton/scripts/script-0-f2m9s\n\ncontainer step-appstudio-summary: \n\nBuild Summary:\n\nBuild repository: https://github.com/redhat-appstudio-qe/devfile-sample-python-basic?rev=47fc22092005aabebce233a9b6eab994a8152bbd\n\nEnd Summary\n",
      }
  In [It] at: /github.com/redhat-appstudio/e2e-tests/tests/build/build_templates.go:200 @ 05/14/24 18:19:17.576
```

We should probably just merge this PR as is. I can't think of how adding a new Task would cause issues on that test

ralphbean commented:
I'm force merging this after Luiz pinged me about it. Neither of us could reason that adding a new task would cause CI to fail in that way.

Build team - if you would rather I not use admin powers to do this in the future, please reach out and let me know and I'll adjust.

lcarva commented:
FWIW, the e2e-tests failures seem to be due to too many secrets linked to the ServiceAccount:

```
$ oc -n build-templates-e2e get sa appstudio-pipeline -o yaml | yq .secrets[].name | wc -l
971
```

I was able to get this in Splunk from one of the Tasks in the failing test pipeline:
```
{"lastTransitionTime":"2024-05-14T18:17:09Z","message":"failed to create task run pod \"devfile-sample-python-basic-fgrv-x6q8m-build-container\": etcdserver: request is too large. Maybe missing or invalid Task build-templates-e2e/","reason":"PodCreationFailed","status":"False","type":"Succeeded"}
```
